src/app/shared/input/input-box/input-box.component.ts

```diff
@@ -52,10 +52,12 @@ export class InputBoxComponent implements OnInit {
   @Input() public textAttribute: boolean = false;
   @Input() public innerClass: string = '';
 
-  @Output() public focus: EventEmitter<void> = new EventEmitter();
+  @Output() public focus: EventEmitter<string> = new EventEmitter();
   @Output() public blur: EventEmitter<void> = new EventEmitter();
   @Output() public newValue: EventEmitter<string> = new EventEmitter();
   @Output() public emptyValue: EventEmitter<void> = new EventEmitter();
+  @Output() public enter = new EventEmitter();
+  @Output() public keyDown = new EventEmitter<KeyboardEvent>();
```

Comment:
Why do we need to emit both `keyDown` and `enter` events from this component? Is it not sufficient to emit just `keyDown` and then check if it is Enter or whatever key in the parent component?

Reply:
There's one problem with this approach. I need to catch enter before the input loses focus. But in the `input-box-component.html` `element.blur()` is called on keydown.enter. That's the reason I put separate event before blur is called.